src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+        theta: Identifier for the gate parameter given in qibo.raw. If None, do gate
+            set tomography with only gate provided by class_qubit_gate.
+        operator_name: Name of gate given in qibo.raw used for saving data.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+
+    Returns:
+        numpy.matrix: Matrix with elements jk equivalent to either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
+    """
+
+    import numpy as np
+
+    from qibo import gates
+    from qibo.backends import GlobalBackend
+    from qibo.models import Circuit
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if class_qubit_gate is None:
+        print("Empty circuit without any operator (1 qb)")
+
+    def GST_measure_1qb(circuit, k, j):
+        idx_basis = ["I", "X", "Y", "Z"]
+
+        result = circuit.execute(nshots=NshotsGST)
+        counts = dict(result.frequencies(binary=True))
+
+        if save_data is not None:
+            import os
+
+            # If location does not exist, create location.
+            if class_qubit_gate is None:
+                filedirectory = "GST_nogates_qibo_1qb/"
+                fname = "No gates (k=%.2d,j=%.2d) %s_basis.txt" % (k, j, idx_basis[j])
+                GST_nogates_qibo_1qb = "GST_nogates_qibo_1qb"
+                if not os.path.exists(GST_nogates_qibo_1qb):
+                    os.makedirs(GST_nogates_qibo_1qb)
+
+            elif class_qubit_gate is not None:
+                filedirectory = "GST_gates_qibo_1qb/"
+                if theta is None:
+                    fname = "%s gate (k=%.2d,j=%.2d) %s_basis.txt" % (
+                        class_qubit_gate,
+                        k,
+                        j,
+                        idx_basis[j],
+                    )
+                elif theta is not None:
+                    fname = "%s(%.12f) gate (k=%.2d,j=%.2d) %s_basis.txt" % (
+                        class_qubit_gate,
+                        theta,
+                        k,
+                        j,
+                        idx_basis[j],
+                    )
+                GST_gates_qibo_1qb = "GST_gates_qibo_1qb"
+                if not os.path.exists(GST_gates_qibo_1qb):
+                    os.makedirs(GST_gates_qibo_1qb)
+
+            # Save counts into a matrix. 0th column: word; 1st column: count value
+            # --------------------------------------------------------------------
+            fout = filedirectory + fname
+            fo = open(fout, "w")
+            for v1, v2 in counts.items():
+                fo.write(str(v1) + "  " + str(v2) + "\n")
+            fo.close()
+
+        # Find value for matrix_jk
+        # -------------------------
+
+        expectation_signs = np.matrix([[1, 1, 1, 1], [1, -1, -1, -1]])
+
+        matrix = np.zeros((2, 2))
+        for ii in range(0, 2):
+            word = bin(ii)[2:]
+            matrix[ii, 0] = word
```

Comment:
So instead of looping
```python
matrix = np.zeros((2, 2))
for ii in range(0,2):
    word = bin(ii)[2:]
    matrix[ii,0] = int(word, 2)
```
would this be better (for single qubit case)?
```python
matrix = np.zeros((2, 2))
matrix[0, 0] = int(bin(0)[2:], 2) 
matrix[1, 0] = int(bin(1)[2:], 2) 
```
and similarly for the 2-qubit case
```python
matrix = np.zeros((4, 2))
matrix[0, 0] = int(bin(0)[2:], 2) 
matrix[1, 0] = int(bin(1)[2:], 2) 
matrix[2, 0] = int(bin(2)[2:], 2) 
matrix[3, 0] = int(bin(3)[2:], 2) 
```

Reply:
I am not really sure why you need all these int(bin()). The matrices are fixed and small anyway, so you could just write them down by hand.
For example for 2 qubits
```
matrix = np.array([[0,0], [1,0], [2,0], [3,0]])
```
or if you prefer
```
np.vstack((np.arange(4), np.zeros(4))).T
```

Reply:
I've removed the loop and implemented this. 
```matrix = np.array([[0,0], [1,0], [10,0], [11,0]])```